packages/ember-glimmer/tests/integration/components/append-test.js

```diff
@@ -35,13 +35,235 @@ class AbstractAppendTest extends RenderingTest {
     this.ids.push(component.elementId);
   }
 
+  ['@test such lifecycle'](assert) {
+    let hooks = [];
+
+    let oldRegisterComponent = this.registerComponent;
+    let componentsByName = {};
+
+    // TODO: refactor/combine with other life-cycle tests
+    this.registerComponent = function(name, _options) {
+      function pushHook(hookName) {
+        hooks.push([name, hookName]);
+      }
+
+      let options = {
+        ComponentClass: _options.ComponentClass.extend({
+          init() {
+            expectDeprecation(() => { this._super(...arguments); }, /didInitAttrs called/);
+            if (name in componentsByName) {
+              throw new TypeError('Component named: ` ' + name + ' ` already registered');
+            }
+            componentsByName[name] = this;
+            pushHook('init');
+            this.on('init', () => pushHook('on(init)'));
+          },
+
+          didInitAttrs(options) {
+            pushHook('didInitAttrs', options);
+          },
+
+          didReceiveAttrs() {
+            pushHook('didReceiveAttrs');
+          },
+
+          willInsertElement() {
+            pushHook('willInsertElement');
+          },
+
+          willRender() {
+            pushHook('willRender');
+          },
+
+          didInsertElement() {
+            pushHook('didInsertElement');
```

Comment:
tracking issue to ensure these do get back ported: https://github.com/emberjs/ember.js/issues/14451